### PR TITLE
NAS-5360 Enforcing api-token permission python sdk update

### PR DESCRIPTION
**Pull request by JKlucka**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/tests/catalog/fixtures/permissions/manage_organization_permissions.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
+# (C) 2023 GoodData Corporation
```
[the remark this answers is not in the document]
Comment (JKlucka) [reply gooddata-sdk/tests/catalog/fixtures/permissions/manage_organization_permissions.yaml:1]: AIO with dev latest, if I did it correctly

File: gooddata-sdk/gooddata_sdk/catalog/permission/service.py [hunk 2]
```diff
@@ -106,3 +108,54 @@ def manage_dashboard_permissions(
...
+            [CatalogDeclarativeOrganizationPermission]:
+                List of all declarative organization permissions.
+        """
+
+        catalog_list = []
+        for permission in self._layout_api.get_organization_permissions():
+            catalog_list.append(CatalogDeclarativeOrganizationPermission.from_api(permission))
+        return catalog_list
+
+    def put_organization_permissions(self, org_permissions: List[CatalogDeclarativeOrganizationPermission]) -> None:
```
[the remark this answers is not in the document]
Comment (JKlucka) [reply gooddata-sdk/gooddata_sdk/catalog/permission/service.py:128]: ok, thanks

File: gooddata-sdk/gooddata_sdk/catalog/permission/service.py [hunk 3]
```diff
@@ -106,3 +108,54 @@ def manage_dashboard_permissions(
             [permission.to_api() for permission in permissions_for_assignee],
             _check_return_type=False,
         )
+
+    def get_organization_permissions(self) -> List[CatalogDeclarativeOrganizationPermission]:
```
[the remark this answers is not in the document]
Comment (JKlucka) [reply gooddata-sdk/gooddata_sdk/catalog/permission/service.py:112]: ok, thanks

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/tests/catalog/fixtures/permissions/manage_organization_permissions.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (JKlucka) [reply gooddata-sdk/tests/catalog/fixtures/permissions/manage_organization_permissions.yaml:1]: I regenerated on AIO 3.0.0 as recommended and the commented header was added manually as I saw it in some other fixture files

File: gooddata-sdk/tests/catalog/test_catalog_permission.py [hunk 4]
```diff
@@ -54,6 +56,12 @@ def _assert_default_permissions(catalog_declarative_permissions: CatalogDeclarat
     assert set(permission.name for permission in catalog_declarative_permissions.permissions) == {"ANALYZE", "VIEW"}
 
 
+def _assert_organization_permissions_id(
+    catalog_organization_permissions: [CatalogDeclarativeOrganizationPermission],
```
Comment (hkad98) [inline gooddata-sdk/tests/catalog/test_catalog_permission.py:60]: This is not correct typing. Note that `mypy` does not work in tests, so it is undetected. Please use `List[CatalogDeclarativeOrganizationPermission]`.

File: gooddata-sdk/gooddata_sdk/catalog/permission/service.py [hunk 5]
```diff
@@ -106,3 +108,56 @@ def manage_dashboard_permissions(
...
+        Args:
+            None
+
+        Returns:
+            [CatalogDeclarativeOrganizationPermission]:
+                List of all declarative organization permissions.
+        """
+
+        catalog_list = []
+        for permission in self._layout_api.get_organization_permissions():
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/catalog/permission/service.py:124]: Please store the result from `self._layout_api.get_organization_permissions()` first and then use it.

e.g.:
`for permission in organization_permissions:`

File: gooddata-sdk/gooddata_sdk/catalog/permission/service.py [hunk 6]
```diff
@@ -106,3 +108,56 @@ def manage_dashboard_permissions(
...
+
+        Args:
+            organization_permission_assignments ([CatalogOrganizationPermissionAssignment]):
+                Object containing List of users and user group and desired organization permissions. Set empty list
+                permissions for user/user group means remove organization permissions.
+        Returns:
+            None
+        """
+        permissions = [permission.to_api() for permission in organization_permission_assignments]
+        print(permissions)
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/catalog/permission/service.py:159]: Please delete this print.

File: gooddata-sdk/gooddata_sdk/catalog/permission/service.py [hunk 7]
```diff
@@ -106,3 +108,56 @@ def manage_dashboard_permissions(
...
+        """Provide managing organization permissions for user and user groups.
+
+        Args:
+            organization_permission_assignments ([CatalogOrganizationPermissionAssignment]):
+                Object containing List of users and user group and desired organization permissions. Set empty list
+                permissions for user/user group means remove organization permissions.
+        Returns:
+            None
+        """
+        permissions = [permission.to_api() for permission in organization_permission_assignments]
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/catalog/permission/service.py:158]: This seems to be unused.

File: gooddata-sdk/tests/catalog/test_catalog_permission.py (hunk 4 above)
Comment (JKlucka) [reply gooddata-sdk/tests/catalog/test_catalog_permission.py:60]: I missed this one, thank you

File: gooddata-sdk/gooddata_sdk/catalog/permission/service.py (hunk 6 above)
Comment (JKlucka) [reply gooddata-sdk/gooddata_sdk/catalog/permission/service.py:159]: forgot to remove, thanks

File: gooddata-sdk/gooddata_sdk/catalog/permission/service.py (hunk 7 above)
Comment (JKlucka) [reply gooddata-sdk/gooddata_sdk/catalog/permission/service.py:158]: right, now to be used as input param for the following line:
`self._actions_api.manage_organization_permissions(permissions, _check_return_type=False,)`

File: gooddata-sdk/tests/catalog/test_catalog_permission.py [hunk 8]
```diff
@@ -164,3 +172,53 @@ def test_list_dashboard_permissions(test_config):
...
+            )
+        ],
+    )
+
+    catalog_declarative_permissions_initial = sdk.catalog_permission.get_declarative_organization_permissions()
+    assert len(catalog_declarative_permissions_initial) == 2
+    assert set(org_permission.name for org_permission in catalog_declarative_permissions_initial) == {
+        "MANAGE",
+        "SELF_CREATE_TOKEN",
+    }
```
Comment (hkad98) [inline gooddata-sdk/tests/catalog/test_catalog_permission.py:224]: I am still determining the tests. There is no AfterAll or AfterEach mechanism as is, for example, known in Java. Therefore, if you put something, then it stays put. There should be a `try-finally` block where the change is reverted in the final block back to the original state.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Review by hkad98 — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR398

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul/214/ : FAILURE in 1m 21s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/364/ : SUCCESS in 1m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/368/ : SUCCESS in 10s

**Pull request merged**